pkg/util/deployment/deployment.go

```diff
@@ -206,15 +206,15 @@ func addHashKeyToRSAndPods(deployment *extensions.Deployment, c clientset.Interf
 	// 2. Update all pods managed by the rs to have the new hash label, so they will be correctly adopted.
 	selector, err := unversioned.LabelSelectorAsSelector(updatedRS.Spec.Selector)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error in converting selector to label selector: %s", err)
```

Comment:
As long as you're doing this, could you please print identifying information in the message, like updatedRS.Name?

Same for the other log messages, to facilitate searching/filtering for relevant log lines.